main/main.go

```diff
@@ -374,6 +374,23 @@ func initSeedGen(ctx context.Context, c client.Client, log *logr.Logger) error {
 		return err
 	}
 
+	// Restore original pull-secret after seed creation.
+	// During seed creation, the pull-secret was removed; it needs to be restored back
+	// to allow the cluster operators to fully recover in the seed cluster.
+	log.Info("Restore original pull-secret after seed creation")
+	dockerConfigJSON, err := os.ReadFile(common.PathOutsideChroot(utils.StoredPullSecret))
```

Comment:
If file doesn't exist, why to fail? Maybe we failed to create it before it? I think we can just skip this part in case utils.StoredPullSecret doesn't exists.

Reply:
How can it be OK?
If you end up without a pull secret it means that something went wrong.

Reply:
Lets say seed creation failed before removing PS, now you will fail to restore it and cleanup phase will get an error.
I assume if file doesn't exists it means seed creator didn't add it and it should be fine to validate if it exists before

Reply:
discussed by chat - this PR doesn't remove the pull-secret, it just updates the secret in the cluster and waits for MCO to update it in the filesystem. There should always be a pull-secret file in the node's filesystem, hence we prefer to fail if the operator doesn't find it during this restoration.